OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1413 @@
+
+        val cf = m.classFile
+        val compType = cf.thisType
+
+        if (project.classHierarchy.isSubtypeOf(compType, ObjectType(androidActivity))) {
+            val relevantLifecycleMethods = ListBuffer.empty[Method]
+            relevantLifecycleMethods ++= cf.findMethod(onStart)
+            relevantLifecycleMethods ++= cf.findMethod(onResume)
+            relevantLifecycleMethods ++= cf.findMethod(onRestart)
+            if (relevantLifecycleMethods.nonEmpty) {
+                callbackMethods.foreach { ce ⇒
+                    if (ce.classFile != m.classFile) {
+                        relevantLifecycleMethods.foreach { lm ⇒
+                            edges += ((lm, 0) -> ce)
+                        }
+                    }
+                }
+                return
+            }
+
+        } else if (project.classHierarchy.isSubtypeOf(compType, ObjectType(androidService))) {
+            val relevantLifecycleMethods = ListBuffer.empty[Method]
+            relevantLifecycleMethods ++= cf.findMethod(onStart)
+            relevantLifecycleMethods ++= cf.findMethod(onBind)
+            relevantLifecycleMethods ++= cf.findMethod(onRebind)
+            if (relevantLifecycleMethods.nonEmpty) {
+                callbackMethods.foreach { ce ⇒
+                    if (ce.classFile != m.classFile) {
+                        relevantLifecycleMethods.foreach { lm ⇒
+                            edges += ((lm, 0) -> ce)
+                        }
+                    }
+                }
+                return
+            }
+        }
+    }
+
+    //Checks if the intent is reachable within the processed method
+    def findOrigin(
+                    params:           Seq[Expr[DUVar[ValueInformation]]],
+                    tacCode:          AITACode[TACMethodParameter, ValueInformation],
+                    originalUseSites: mutable.Map[Int, List[Int]]
+                  ): Boolean = {
+        params.foreach { p ⇒
+            if (p.asVar.value.toString.startsWith(intentStart.head) || p.asVar.value.toString.startsWith(intentStart(1))) {
+                //the intent to its source
+                val defIndex = p.asVar.definedBy.head
+                if (defIndex < 0) {
+                    return false //intent is not reachable
+                }
+                if (originalUseSites.exists(k ⇒ k._2.contains(defIndex))) {
+                    return true
+                }
+                val stm = tacCode.stmts(defIndex)
+                if (stm.isAssignment) {
+                    if (stm.asAssignment.expr.isStaticFunctionCall || stm.asAssignment.expr.isVirtualFunctionCall) {
+                        return findOrigin(stm.asAssignment.expr.asFunctionCall.params, tacCode, originalUseSites)
+                    }
+                }
+            }
+        }
+        false //intent is not reachable
+    }
+
+    /**
+     * searches the parameters of functionCall to find the origin of the intent and return the class and method
```

Comment:
Please format your ScalaDoc comments: Descriptions should start with a captial letter (as any proper sentence) and end with a full stop. Remove blank lines that are unnecessary.